includes/class-amp-theme-support.php

```diff
@@ -506,13 +506,25 @@ public static function get_amp_component_scripts() {
 	 * Start output buffering.
 	 */
 	public static function start_output_buffering() {
-		ob_start( array( __CLASS__, 'finish_output_buffering' ) );
+		ob_start( array( __CLASS__, 'finish_buffer_add_header' ) );
 	}
 
 	/**
-	 * Finish output buffering.
+	 * Get the result of the output buffering, and add a header.
 	 *
 	 * @todo Do this in shutdown instead of output buffering callback?
+	 * @param string $output Buffered output.
+	 * @return string Finalized output.
+	 */
+	public static function finish_buffer_add_header( $output ) {
+		$markup = self::finish_output_buffering( $output );
+		AMP_Mutation_Utils::add_header();
```

Comment:
We'll want to limit this to only be added when a user specifically requests this additional information. Like there should be a nonce that must be present to authorize the reporting.

Reply:
Thanks, @westonruter. That's a good idea to add the header only for users with a nonce.

Reply:
Applied with a nonce, details to come shortly.

Reply:
`AMP_Mutation_Utils::add_header()` [now has a check for authorization](https://github.com/Automattic/amp-wp/pull/912/files#diff-60304ef5c3b3f17aeb304813b8716ebcR287) via `self::is_authorized()`. That verifies the nonce and capability.